### DISCUSSION TITLE
Rust plugin

### DISCUSSION THREAD

**Discussion by hverlin** [jdx/mise#2932] [General]
On https://mise.jdx.dev/lang/rust.html
>  In fact, I don't think you should actually use mise for rust development.

Do you mean that you discourage using the `rust asdf` plugin? Seems that `mise` is still useful in rust projects for `tasks` or `env` (or the project requires other tools).

In fact, I think https://github.com/jdx/demand/blob/main/.mise.toml is a good example of how someone could use `mise` in a rust project?

My suggestion would be to update the documentation to indicate that you can use `mise` for rust projects, but not necessarily the `rust` tool (asdf rust). Something like:

- Using `rustup`. Set `mise settings set disable_tools rust` so that mise never install `rust` and install `rust` using `rustup`
- Other way: use the rust `asdf` plugin

I am also wondering what it would mean for `mise` to implement a core `rust` plugin like the documentation suggests. Would it be a breaking change in case someone was relying on the `rust` asdf plugin?

**Comment by MrSquaare** [comment]
I agree with this opinion and would add one more thing, I installed mise because I want to have a single tool to manage my SDKs and their versions, regardless of whether an official tool exists or a popular tool (nvm, pyenv) does the same thing.

This also allows me to create projects for which the only prerequisite is to have mise installed. Developers will only have to do a `mise install` to setup their dev environment.

**Comment by hverlin** [comment]
Seems that the https://mise.jdx.dev/lang/rust.html#rust has been updated now 🎉 
Thank you, @jdx!
